### PR TITLE
[New Rule] Remote File Download via Powershell

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/659

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/command_and_control_remote_file_copy_powershell.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+type = "eql"
+
+query = '''
+
+sequence by host.id, process.entity_id with maxspan = 30s
+  [network where process.name : "powershell.exe" and network.protocol == "dns" and
+   not dns.question.name : ("localhost", "*.microsoft.com", "*.azureedge.net", "*.powershellgallery.com", "*.windowsupdate.com", "metadata.google.internal") and 
+   not user.domain : "NT AUTHORITY"]
+  [file where event.type == "creation" and file.extension : ("exe", "dll", "ps1", "bat") and 
+   not file.path : "C:\\Windows\\Temp\\__PSScriptPolicy*.ps1"]
```
Comment (brokensound77) [inline rules/windows/command_and_control_remote_file_copy_powershell.toml:29]:
```suggestion
sequence by host.id, process.entity_id with maxspan=30s
  [network where process.name : "powershell.exe" and network.protocol == "dns" and
   not dns.question.name : ("localhost", "*.microsoft.com", "*.azureedge.net", "*.powershellgallery.com", "*.windowsupdate.com", "metadata.google.internal") and 
   not user.domain : "NT AUTHORITY"]
  [file where event.type == "creation" and file.extension : ("exe", "dll", "ps1", "bat") and 
   not file.path : "C:\\Windows\\Temp\\__PSScriptPolicy*.ps1"]
```

Those leading wildcards for `dns.question.name` are not great, but I don't think there is any better way atm

File: rules/windows/command_and_control_remote_file_copy_powershell.toml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+[rule]
+author = ["Elastic"]
+description = """
+Identifies powershell.exe being used to download an executable file from a remote destination.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Remote File Download via Powershell"
```
Comment (brokensound77) [inline rules/windows/command_and_control_remote_file_copy_powershell.toml:15]:
```suggestion
name = "Remote File Download via PowerShell"
```

File: rules/windows/command_and_control_remote_file_copy_powershell.toml (hunk 1 above)
Comment (Samirbous) [reply rules/windows/command_and_control_remote_file_copy_powershell.toml:29]: yeah, its noise vs performance for powershell download activity

File: rules/windows/command_and_control_remote_file_copy_powershell.toml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+author = ["Elastic"]
+description = """
+Identifies powershell.exe being used to download an executable file from a remote destination.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Remote File Download via PowerShell"
+risk_score = 43
```
Comment (bm11100) [inline rules/windows/command_and_control_remote_file_copy_powershell.toml:16]:
```suggestion
risk_score = 47
```

**Review by bm11100 — APPROVED** [review]

File: rules/windows/command_and_control_remote_file_copy_powershell.toml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Command and Control"]
+type = "eql"
+
+query = '''
+sequence by host.id, process.entity_id with maxspan=30s
+  [network where process.name : "powershell.exe" and network.protocol == "dns" and
+   not dns.question.name : ("localhost", "*.microsoft.com", "*.azureedge.net", "*.powershellgallery.com", "*.windowsupdate.com", "metadata.google.internal") and 
+   not user.domain : "NT AUTHORITY"]
+  [file where event.type == "creation" and file.extension : ("exe", "dll", "ps1", "bat") and
```
Comment (seth-goodwin) [inline rules/windows/command_and_control_remote_file_copy_powershell.toml:27]:
```suggestion
    [file where process.name : "powershell.exe" and event.type == "creation" and file.extension : ("exe", "dll", "ps1", "bat") and 
```
After some discussion with @brokensound77, we think the addition of the `process.name` value to the file event might tighten up the sequencing to reduce load.

File: rules/windows/command_and_control_remote_file_copy_powershell.toml [hunk 5]
```diff
@@ -0,0 +1,56 @@
+[metadata]
+creation_date = "2020/11/30"
+maturity = "production"
+updated_date = "2020/11/30"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies powershell.exe being used to download an executable file from a remote destination.
```
Comment (seth-goodwin) [inline rules/windows/command_and_control_remote_file_copy_powershell.toml:9]:
```suggestion
Identifies powershell.exe being used to download an executable file from an untrusted remote destination.
```
The logic implies a list of destinations that are trusted for this activity.

**Review by seth-goodwin — APPROVED** [review]
LGTM 👍

**Pull request merged**